internal/parse/parse.go

```diff
@@ -0,0 +1,377 @@
+			// TODO: detect duplication of these options.
+			// (Is this necessary?)
+			newMount.Options = append(newMount.Options, kv[0])
+		case "rw", "readwrite":
+			newMount.Options = append(newMount.Options, "rw")
+		case "readonly", "ro":
+			// Alias for "ro"
+			newMount.Options = append(newMount.Options, "ro")
+			setReadOnly = true
+		case "shared", "rshared", "private", "rprivate", "slave", "rslave", "Z", "z", "U":
+			newMount.Options = append(newMount.Options, kv[0])
+			setShared = true
+		case "bind-propagation":
+			if len(kv) == 1 {
+				return newMount, "", errors.Wrapf(optionArgError, kv[0])
+			}
+			newMount.Options = append(newMount.Options, kv[1])
+		case "id":
+			id = kv[1]
+		case "from":
+			fromImage = kv[1]
+		case "target", "dst", "destination":
+			if len(kv) == 1 {
+				return newMount, "", errors.Wrapf(optionArgError, kv[0])
+			}
+			if err := parse.ValidateVolumeCtrDir(kv[1]); err != nil {
+				return newMount, "", err
+			}
+			newMount.Destination = kv[1]
+			setDest = true
+		case "src", "source":
+			if len(kv) == 1 {
+				return newMount, "", errors.Wrapf(optionArgError, kv[0])
+			}
+			newMount.Source = kv[1]
+		case "mode":
+			mode, err = strconv.ParseUint(kv[1], 8, 32)
+			if err != nil {
+				return newMount, "", errors.Wrapf(err, "Unable to parse cache mode")
+			}
+		case "uid":
+			uid, err = strconv.Atoi(kv[1])
+			if err != nil {
+				return newMount, "", errors.Wrapf(err, "Unable to parse cache uid")
+			}
+		case "gid":
+			gid, err = strconv.Atoi(kv[1])
+			if err != nil {
+				return newMount, "", errors.Wrapf(err, "Unable to parse cache gid")
+			}
+		default:
+			return newMount, "", errors.Wrapf(errBadMntOption, kv[0])
+		}
+	}
+
+	if !setDest {
+		return newMount, "", noDestError
+	}
+
+	if fromImage != "" {
+		// do not create cache on host
+		// instead use read-only mounted image as cache
+		mountPoint := ""
+		//TODO: remove this selinux check when comment is resolved. https://github.com/containers/buildah/pull/3590#issuecomment-956349109
+		if additionalMountPoints != nil && (selinux.EnforceMode() != 1) {
+			if val, ok := additionalMountPoints[fromImage]; ok {
+				mountPoint = val
+			}
+		}
+		// Cache does not supports using image so if not stage found
+		// return with error
+		if mountPoint == "" {
+			return newMount, "", fmt.Errorf("no stage or image found with name %s", fromImage)
+		}
+		newMount.Options = append(newMount.Options, "rw")
+		if strings.HasPrefix(newMount.Source, ".") || newMount.Source == "" || !filepath.IsAbs(newMount.Source) || fromImage != "" {
+			// path should be /contextDir/specified path
+			newMount.Source = filepath.Join(mountPoint, filepath.Clean(string(filepath.Separator)+newMount.Source))
+		}
+	} else {
+		// we need to create cache on host if no image is being used
+
+		// since type is cache and cache can be reused by consecutive builds
+		// create a common cache directory, which should persists on hosts within temp lifecycle
```

Comment:
Typo: "which should persists" should be either "which should persist" or "which persists"